databases/functions/postAnonUserInfo/check_self_username_in_post.js

```diff
@@ -13,19 +12,33 @@ const UsersFunction = require('../users');
  * @param {Model} userModel
  * @param {CheckSelfUsernameInPostParams} data
  * @param {Transaction} transaction
+ * @param {boolean} is_anonymous
  * @returns
  */
-module.exports = async (postAnonUserInfoModel, userModel, data, transaction = null) => {
+module.exports = async (
+  postAnonUserInfoModel,
+  userModel,
+  data,
+  transaction = null,
+  is_anonymous = true
+) => {
   const {userId, postId} = data;
 
-  const anonymousUser = await UsersFunction.findAnonymousUserId(userModel, userId);
-  console.log('anonymousUser');
-  console.log(anonymousUser);
+  let anon_user_id;
+  if (!is_anonymous) {
+    const anonymousUser = await UsersFunction.findAnonymousUserId(userModel, userId);
+    console.log('anonymousUser');
+    console.log(anonymousUser);
+    anon_user_id = anonymousUser?.user_id;
+  } else {
+    anon_user_id = userId;
+  }
+
   let postAnonUserInfo = await postAnonUserInfoModel.findOne(
     {
       where: {
         post_id: postId,
-        anon_user_id: anonymousUser?.user_id
+        anon_user_id
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for handling the `is_anonymous` flag seems to be reversed. If `is_anonymous` is `true`, we should find the anonymous user ID, not use the provided `userId`. Conversely, if `is_anonymous` is `false`, we should use the provided `userId` directly.

Also, it's not a good practice to leave console logs in production code. They can expose sensitive information and clutter the output. Consider using a logging library that can be configured to log only in development environments or based on log levels.

```diff
-  let anon_user_id;
-  if (!is_anonymous) {
-    const anonymousUser = await UsersFunction.findAnonymousUserId(userModel, userId);
-    console.log('anonymousUser');
-    console.log(anonymousUser);
-    anon_user_id = anonymousUser?.user_id;
-  } else {
-    anon_user_id = userId;
-  }
+  let anon_user_id = is_anonymous 
+    ? (await UsersFunction.findAnonymousUserId(userModel, userId))?.user_id
+    : userId;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->